server/startup.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+PORT=3000
+
+echo pwd
+cd core
+flutter pub get
+
+cd ../model
+flutter pub get && flutter pub run build_runner build --delete-conflicting-outputs
+
+cd ..
+flutter pub get && flutter pub run intl_generator:extract_to_arb --output-dir=./lib/l10n lib/main/localizations/app_localizations.dart
+
+flutter pub get && flutter pub run intl_generator:generate_from_arb --output-dir=lib/l10n --no-use-deferred-loading lib/main/localizations/app_localizations.dart lib/l10n/intl*.arb
+
+flutter build web --dart-define=SERVER_URL='https://dev.open-paas.org'
```

Comment:
That's an other matter but we shouldn't hardcode the server url :)

Reply:
yes, I am finding the way to use env file instead

Reply:
Building the flutter web app every time you start the SDK is not a best practice.

We should build the web app outside of docker, then only import the resulting files within docker.

Reply:
> We should build the web app outside of docker

How to do it?

Reply:
I'm not sure what the target looks like after being built honestly... might not be that easy. We need to check a bit deeper I think before taking this kind of decision

Reply:
How about, we build it on CI/CD, then import it and build docker image in CI/CD, then publish it to linagora docker hub?

Reply:
@tnstung Can you give me some idea for this?

Reply:
You can inject env variables when running a docker container btw (the link I posted in the other comment, you can see it too: https://rebar-ahmad.medium.com/env-aware-flutter-apps-in-docker-b899d0deebfd)

I stumbled upon this flutter package also if you prefer using env file: https://pub.dev/packages/flutter_dotenv

From all the articles and doc I could read though, it's obvious that the env variable injection is only possible during the build phase. It seems the best way if you want your docker image to be env custom is to build it inside the image. 

@chibenwa but is that really an issue? Maybe not best practice, but with this limitation on flutter webapp, this might still be the best choice.

It is easier to maintain like this than changing constantly your CI conf depending on how many environments you want to serve. If you get multiples customers with different domains in the future, it could become very chaotic and error prone to maintain for example

Reply:
Fair.

But then should't we consider multi stage builds? https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds

Rationals: docker images should not be big for convenience and security resons.

We could have a docker file that first build the static assets then just copy the build result to a nice minimal nginx image.

Reply:
Forgot the multi stage build but yeah very good suggestion!